File: union-find/python/union-find/src/QF.py
# coding: utf8

'''
Created on 25 בינו 2015

@author: benjaminbo
'''

class QF(object):
    '''
    quickFind
    '''
    id = None

    def __init__(self, N):
        self.id = list(range(N))
        
    def connected(self, p, q):
        return self.id[p] == self.id[q]
    
    def union(self, p, q):
        idp = self.id[p]
        idq = self.id[q]
        
        for i in range(len(self.id)):
            if self.id[i] == idp : self.id[i] = idq
    

File: union-find/java/unionfind/src/unionfind/QF.java
package unionfind;

public class QF {

	private int[] id;

	public QF(int N) {
		id = new int[N];
		for (int i = 0; i < id.length; i++)
			id[i] = i;
	}

	public boolean connected(int p, int q) {
		return id[p] == id[q];
	}

	public void union(int p, int q) {
		// p recupere l identite de q
		int idp = id[p];
		int idq = id[q];

		for (int i = 0; i < id.length; i++)
			if (id[i] == idp)
				id[i] = idq;
	}
}
